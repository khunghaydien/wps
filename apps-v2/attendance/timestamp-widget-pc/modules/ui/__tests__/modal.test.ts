import snapshotDiff from 'snapshot-diff';

import reducer, {
  actions,
  CLOSE_MODAL,
  initialState,
  SHOW_MODAL,
} from '../modal';

describe('reducer()', () => {
  describe('@init', () => {
    it('@@init', () => {
      // @ts-ignore
      const next = reducer(undefined, { type: '@@INIT' });
      expect(snapshotDiff({}, next)).toMatchSnapshot();
    });
  });
  describe(`${SHOW_MODAL}`, () => {
    it('SHOW_MODAL', () => {
      const next = reducer(initialState, actions.showModal(45));
      expect(snapshotDiff(initialState, next)).toMatchSnapshot();
    });
  });
  describe(`${CLOSE_MODAL}`, () => {
    it('CLOSE_MODAL', () => {
      const state = { insufficientRestTime: '45', isShowModal: true };
      const next = reducer(initialState, actions.closeModal());
      expect(snapshotDiff(state, next)).toMatchSnapshot();
    });
  });
});
