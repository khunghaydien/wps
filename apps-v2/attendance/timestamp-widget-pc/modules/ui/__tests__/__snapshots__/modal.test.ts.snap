// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer() @init @@init 1`] = `
"Snapshot Diff:
- First value
+ Second value

- Object {}
+ Object {
+   \\"insufficientRestTime\\": 0,
+   \\"isShowModal\\": false,
+ }"
`;

exports[`reducer() TIMESTAMP_WIDGET_CLOSE_MODAL CLOSE_MODAL 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
-   \\"insufficientRestTime\\": \\"45\\",
-   \\"isShowModal\\": true,
+   \\"insufficientRestTime\\": 0,
+   \\"isShowModal\\": false,
  }"
`;

exports[`reducer() TIMESTAMP_WIDGET_SHOW_MODAL SHOW_MODAL 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
-   \\"insufficientRestTime\\": 0,
-   \\"isShowModal\\": false,
+   \\"insufficientRestTime\\": 45,
+   \\"isShowModal\\": true,
  }"
`;
